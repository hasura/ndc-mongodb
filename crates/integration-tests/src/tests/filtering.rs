use insta::assert_yaml_snapshot;

use crate::graphql_query;

#[tokio::test]
async fn filters_on_extended_json_using_string_comparison() -> anyhow::Result<()> {
    // Skip this test in MongoDB 5 because the example fails there. We're getting an error:
    //
    // > Kind: Command failed: Error code 5491300 (Location5491300): $documents' is not allowed in user requests, labels: {}
    //
    // This doesn't affect native queries that don't use the $documents stage.
    if let Ok(image) = std::env::var("MONGODB_IMAGE") {
        if image == "mongo:5" {
            return Ok(());
        }
    }

    assert_yaml_snapshot!(
        graphql_query(
            r#"
                query Filtering {
                  extendedJsonTestData(where: { value: { _regex: "hello" } }) {
                    type
                    value
                  }
                }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}
