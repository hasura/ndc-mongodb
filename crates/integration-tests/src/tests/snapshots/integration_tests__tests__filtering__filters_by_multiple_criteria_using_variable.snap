---
source: crates/integration-tests/src/tests/filtering.rs
expression: "graphql_query(r#\"\n            query TestQuery($filter: [MoviesOrderBy!] = {}) {\n              movies(\n                order_by: $filter\n                where: { year: { _gt: 0 }, imdb: { rating: { _gt: 9 } } }\n                limit: 15\n              ) {\n                title\n                year\n                rated\n              }\n            }\n            \"#).variables(json!({\n    \"filter\": [{ \"year\": \"Desc\" }, { \"rated\": \"Desc\" },]\n})).run().await?"
---
data:
  movies:
    - title: "A Brave Heart: The Lizzie Velasquez Story"
      year: 2015
      rated: PG-13
    - title: The Real Miyagi
      year: 2015
      rated: ~
    - title: Over the Garden Wall
      year: 2014
      rated: TV-PG
    - title: Frozen Planet
      year: 2011
      rated: ~
    - title: Human Planet
      year: 2011
      rated: ~
    - title: Life
      year: 2009
      rated: PG
    - title: Planet Earth
      year: 2006
      rated: TV-G
    - title: Band of Brothers
      year: 2001
      rated: TV-MA
    - title: The Blue Planet
      year: 2001
      rated: ~
    - title: Pride and Prejudice
      year: 1995
      rated: ~
    - title: Baseball
      year: 1994
      rated: TV-PG
    - title: The Shawshank Redemption
      year: 1994
      rated: R
    - title: The Shawshank Redemption
      year: 1994
      rated: R
    - title: The Civil War
      year: 1990
      rated: ~
    - title: The Civil War
      year: 1990
      rated: ~
errors: ~
